### PR TITLE
Fix 2 runtime errors; issue of failed training accuracy

### PR DESCRIPTION
**Pull request by littlepretty**
* Use isinstance() instead of type(); type() fails to recognize
    tensor.cuda.TensorFloat when mode=gpu, at least on python 2.7.
* Training loss is not a list, after copied back to cpu.
* **Question to author**: on our GPU server, when I ./run_DGCNN.sh, using default settings shown below, training loss decreased from 0.69 to 0.48, but accuracy is constantly 0.00. Any idea why?

```
====== begin of s2v configuration ======
| msg_average = 0
======   end of s2v configuration ======
Namespace(batch_size=50, data='DD', dropout=True, feat_dim=0, fold=10, gm='DGCNN', hidden=128, latent_dim=[32, 32, 32, 1], learning_rate=1e-05, max_lv=4, mode='gpu', num_class=0, num_epochs=200, out_dim=0, seed=1, sortpooling_k=0.6, test_number=0)
loading data
# classes: 2
# maximum node tag: 82
# train: 1061, # test: 117
k used in SortPooling is: 291
Initializing DGCNN
loss: 0.68376 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.45batch/s]
average training of epoch 0: loss 0.69044 acc 0.00000
loss: 0.68515 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.01batch/s]
average test of epoch 0: loss 0.69192 acc 0.00000
loss: 0.70323 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.41batch/s]
average training of epoch 1: loss 0.68251 acc 0.00000
loss: 0.68006 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 46.59batch/s]
average test of epoch 1: loss 0.69230 acc 0.00000
loss: 0.68318 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.63batch/s]
average training of epoch 2: loss 0.67538 acc 0.00000
loss: 0.67748 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.12batch/s]
average test of epoch 2: loss 0.69416 acc 0.00000
loss: 0.66298 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 28.49batch/s]
average training of epoch 3: loss 0.67310 acc 0.00000
loss: 0.67546 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 46.95batch/s]
average test of epoch 3: loss 0.69507 acc 0.00000
loss: 0.62291 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.98batch/s]
average training of epoch 4: loss 0.67112 acc 0.00000
loss: 0.67422 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.14batch/s]
average test of epoch 4: loss 0.69502 acc 0.00000
loss: 0.71296 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.61batch/s]
average training of epoch 5: loss 0.67134 acc 0.00000
loss: 0.67313 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.05batch/s]
average test of epoch 5: loss 0.69503 acc 0.00000
loss: 0.59624 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 27.75batch/s]
average training of epoch 6: loss 0.66957 acc 0.00000
......
average training of epoch 196: loss 0.47397 acc 0.00000
loss: 0.65063 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.11batch/s]
average test of epoch 196: loss 0.58036 acc 0.00000
loss: 0.42728 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 30.07batch/s]
average training of epoch 197: loss 0.47682 acc 0.00000
loss: 0.65348 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.13batch/s]
average test of epoch 197: loss 0.58178 acc 0.00000
loss: 0.45303 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 29.32batch/s]
average training of epoch 198: loss 0.48025 acc 0.00000
loss: 0.65295 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 47.14batch/s]
average test of epoch 198: loss 0.57904 acc 0.00000
loss: 0.36955 acc: 0.00000: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:00<00:00, 29.37batch/s]
average training of epoch 199: loss 0.47433 acc 0.00000
loss: 0.65193 acc: 0.00000: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 46.92batch/s]
average test of epoch 199: loss 0.57938 acc 0.00000

```

### CODE REVIEW

**Comment by muhanzhang** [comment]
Hi, thank you for the several fixes which are necessary for updating the software to pytorch 0.4.0. I further fix the acc=0 issue by calling the .item() method to get python number from scalar tensors. It should work now.
